src/python/pants/backend/python/macros/caof_utils.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import Any, Dict, Optional, Tuple, Union
+
+from packaging.utils import canonicalize_name as canonicalize_project_name
+
+from pants.engine.target import InvalidFieldException
+from pants.util.collections import ensure_str_list
+
+OVERRIDES_TYPE = Optional[Dict[Union[str, Tuple[str, ...]], Dict[str, Any]]]
+
+
+def flatten_overrides_to_dependency_field(
+    overrides_value: OVERRIDES_TYPE, *, macro_name: str, build_file_dir: str
+) -> dict[str, list[str]]:
+    """Flatten `overrides` by ensuring that only `dependencies` is specified."""
+
+    result: dict[str, list[str]] = {}
+    for maybe_key_or_keys, override in (overrides_value or {}).items():
+        keys = (maybe_key_or_keys,) if isinstance(maybe_key_or_keys, str) else maybe_key_or_keys
+        for _raw_key in keys:
+            key = canonicalize_project_name(_raw_key)
```

Comment:
It's slightly weird that the name of this file and this function imply that this is generically useful for CAOFs of all kinds, but in fact this line seems to give away that it's really only for macros that generate python_requirement?

Reply:
Sure, although this is temporary once we replace these w/ target generators, so I'm not that concerned. Also note it's in `python/macros`